app/src/main/java/org/jellyfin/androidtv/data/source/JsonFileSource.kt

```diff
@@ -0,0 +1,33 @@
+package org.jellyfin.androidtv.data.source
+
+import android.content.Context
+import org.jellyfin.apiclient.serialization.GsonJsonSerializer
+import timber.log.Timber
+import java.io.FileNotFoundException
+
+open class JsonFileSource<T>(
+	private val context : Context,
+	private val fileName : String,
+	private val serializer : GsonJsonSerializer,
+	private val clazz : Class<T>
+) {
+	fun read() : T? {
+		try {
+			context.openFileInput(fileName).use {
+				val data = it.readBytes()
+					.toString(Charsets.UTF_8)
+
+				return serializer.DeserializeFromString(data, clazz)
```

Comment:
FYI; The next apiclient version doesn't provide a serializer anymore, and if it did it would not use GSON but kotlinx.serialization. I'd recommend switching to kotlinx.serialization in this PR for all new JSON stuff.

Optionally; it also supports [CBOR](http://cbor.io) which might have slightly better performance.